payjoin/src/receive/optional_parameters.rs

```diff
@@ -4,10 +4,7 @@ use std::fmt;
 use bitcoin::FeeRate;
 use log::warn;
 
-#[cfg(feature = "v2")]
-pub(crate) const SUPPORTED_VERSIONS: [usize; 2] = [1, 2];
-#[cfg(not(feature = "v2"))]
-pub(crate) const SUPPORTED_VERSIONS: [usize; 1] = [1];
+pub(crate) const SUPPORTED_VERSIONS: &[usize] = if cfg!(feature = "v2") { &[1, 2] } else { &[1] };
```

Comment:
I realize now that switching this on cfg is insufficient, since a compiled version may either be running a v1 or v2 receiver, depending on which functions are used. Those functions are going to have to pass supported versions instead. Perhaps there will be separate payjoin::v1, payjoin::v2 modules where you'd only have one of the two public in practice but v1 gets conditionally compiled as pub(crate) for use when v2 is active.